### PR TITLE
feat(/elk): remove /elk need

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Clean up urls in the user's browser

## TODO
- [x] Remove elk from the url path (note this requires setting the elk base path to `/`)
- [ ] Update elk to not use `/elk` for deployment  https://github.com/MozillaSocial/elk/pull/11

### CODE REVIEW

File: haproxy.cfg [hunk 1]
```diff
@@ -44,11 +46,8 @@ backend elk
   server elk "${ELK_SERVICE_AND_PORT-elk.elk-stage.svc.cluster.local:8080}" maxconn "${MAXCONN_ELK-50}"
```
Comment (cvalaas) [inline haproxy.cfg]:
```suggestion
```

File: haproxy.cfg [hunk 2]
```diff
@@ -44,11 +46,8 @@ backend elk
...
```
Comment (cvalaas) [inline haproxy.cfg:66]:
```suggestion
```

File: haproxy.cfg [hunk 3]
```diff
@@ -44,11 +46,8 @@ backend elk
...
 
 backend rewrite-to-elk-plain
```
Comment (cvalaas) [inline haproxy.cfg]:
```suggestion
```

File: haproxy.cfg [hunk 4]
```diff
@@ -44,11 +46,8 @@ backend elk
...
-  http-request replace-path (.*) /elk\1
-  http-request replace-path /elk/publish(.*) /elk/compose\1
```
Comment (cvalaas) [inline haproxy.cfg:48]: This line needs to stay, but without the `/elk`s

File: haproxy.cfg [hunk 5]
```diff
@@ -44,11 +46,8 @@ backend elk
...
-  http-request replace-path /elk/settings/.* /elk/settings
```
Comment (cvalaas) [inline haproxy.cfg:49]: Same here

File: haproxy.cfg [hunk 6]
```diff
@@ -44,11 +46,8 @@ backend elk
...
-  http-request redirect prefix "https://${ROUTER_HOSTNAME-stage.moztodon.nonprod.webservices.mozgcp.net}"
+  server elk "${ELK_SERVICE_AND_PORT-elk.elk-stage.svc.cluster.local:8080}" maxconn "${MAXCONN_ELK-50}"
 
 backend rewrite-to-elk-svrname
   http-request replace-path (.*) "/elk/${MASTODON_HOSTNAME-stage.moztodon.nonprod.webservices.mozgcp.net}\1"
```
Comment (cvalaas) [inline haproxy.cfg]:
```suggestion
  http-request replace-path (.*) "/${MASTODON_HOSTNAME-stage.moztodon.nonprod.webservices.mozgcp.net}\1"
```

**Review by cvalaas — CHANGES_REQUESTED** [review]

**Comment by bassrock** [comment]
@cvalaas Made some updates based on me running locally and with Elk in single instance mode.

**Review by cvalaas — COMMENTED** [review]
Seems fine to me, but will break most of baku's tests. I've added him as a reviewer

File: haproxy.cfg [hunk 7]
```diff
@@ -21,9 +21,12 @@ frontend router
...
-  use_backend elk if { path_beg /elk }
-  use_backend rewrite-to-elk-svrname if { path '/explore' } || { path '/list' } || { path '/local' } || { path '/notifications' } || { path '/public/local' } || { path_beg /@ } || { path_beg /tags }
-  use_backend rewrite-to-elk-plain if { path / } || { path /bookmarks } || { path /conversations } || { path /favourites } || { path /home } || { path /public } || { path /publish } || { path '/search' } || { path_beg /settings }
+  use_backend elk if { path_beg /_nuxt }
+  use_backend elk if { path_beg /api/${ROUTER_HOSTNAME-stage.moztodon.nonprod.webservices.mozgcp.net} }
+  use_backend mastodon if { path_beg /api }
+  use_backend rewrite-publish if { path '/publish' } || { path_beg /settings/ }
+  use_backend elk if { path / } || { path '/compose' } || { path /settings } || { path /bookmarks } || { path '/conversations' } || { path '/favourites' } || { path '/home' } || { path '/public' } || { path '/search' }
+  use_backend elk if { path_beg /settings }  || { path '/explore' } || { path '/list' } || { path '/local' } || { path '/notifications' } || { path '/public/local' } || { path_beg /@ } || { path_beg /tags }
```
Comment (cvalaas) [inline haproxy.cfg]:
```suggestion
  use_backend rewrite-to-elk-svrname if { path '/explore' } || { path '/list' } || { path '/local' } || { path '/notifications' } || { path '/public/local' } || { path_beg /@ } || { path_beg /tags }
```
Guessing `elk` as the backend was a typo?

Comment (bassrock) [reply haproxy.cfg]: HAProxy didnt like a line longer then 240 characters 🤣  this was on purpose.

Comment (cvalaas) [reply haproxy.cfg]: ah, so the `backend rewrite-to-elk-svrname` section can also go away?

Comment (bassrock) [reply haproxy.cfg]: Ah yep. let me burn that.

Comment (bassrock) [reply haproxy.cfg]: Done.

**Comment by bassrock** [comment]
@cvalaas Just added some more, specifically some forward headers that were needed

**Review by bakulf — COMMENTED** [review]
@cvalaas can we run the tests on this PR? (or in general for every PRs?)

**Comment by bassrock** [comment]
@bakulf oh yes we should. We could prob setup the mozsocial docs docker stuff to run integration tests or something in github actions.

**Comment by cvalaas** [comment]
I've been running the tests you wrote, @bakulf , against the stage instance. But, obviously we need to deploy first for that to happen. A bit of a catch-22.
I think I could set up the deploy to run to run the tests after deploying the code, and then maybe rollback the deploy if it fails?

**Comment by bassrock** [comment]
@cvalaas ~FYI im working on updating the tests for this version and running them locally.~ Updated!

**Review by bakulf — APPROVED** [review]

**Review by cvalaas — APPROVED** [review]

**Pull request merged**